src/main/java/seedu/address/model/expense/Category.java

```diff
@@ -0,0 +1,63 @@
+package seedu.address.model.expense;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.ArrayList;
+
+//@@author Jiang Chen
+/**
+ * Represent a category of expenses.
+ * Guarantee: details are present and not null, filed values are validated, immutable.
+ */
+public class Category {
+    public static final String MESSAGE_CATEGORY_CONSTRAINTS =
+            "Category can take any values, and it should not be blank";
+
+    public static final String CATEGORY_VALIDATION_REGEX = "\\p{Alnum}+";
+
+
+    public final String categoryName;
+
+    public final ArrayList<Expense> expenseList = new ArrayList<>();
+
+    /**
+     * Constructs an {@code Category}.
+     *
+     * @param category A valid category.
+     */
+    public Category(String category) {
+        requireNonNull(category);
+        checkArgument(isValidCategory(category), MESSAGE_CATEGORY_CONSTRAINTS);
+        categoryName = category;
```

Comment:
Please use `this` to reference instance variables to avoid confusion between instance and local variables